Modules/_decimal/_decimal.c

```diff
@@ -4746,7 +4954,7 @@ dec_mpd_qquantize(PyObject *v, PyObject *args, PyObject *kwds)
         }
     }
 
-    CONVERT_BINOP_RAISE(&a, &b, v, w, context);
+    CONVERT_BINOP_RAISE(&a, &b, self, w, context);
```

Comment:
You could use `self as v: self` to reduce the diff here & above with the `dec` cases. It's marginal though, it's only a few lines in each case.

Reply:
It does, for exactly this purpose! See the [self converter](https://devguide.python.org/development-tools/clinic/#how-to-use-the-self-converter) section:

> However, you can override Argument Clinic’s converter and specify one yourself. Just add your own self parameter as the first parameter in a block, and ensure that its converter is an instance of `self_converter` or a subclass thereof.
>
> What’s the point? This lets you override the type of `self`, or give it a different default name.